daccs_client/client.py

```diff
@@ -31,94 +25,96 @@ def __init__(self, fallback: Optional[bool] = True) -> None:
             raise RuntimeError(
                 "Could not parse JSON returned from the cloud registry. "
-                "Consider re-trying with 'fallback' set to True when instantiating the DACCS object."
+                f"Consider re-trying with 'fallback' set to True when instantiating the {self.__class__.__name__} "
+                "object."
             )
         self._save_registry_as_cache()
 
     def _load_registry_from_cache(self):
         try:
-            with open(CACHE, "r") as f:
+            with open(CACHE_FNAME, "r") as f:
                 self._registry = json.load(f)
-        except FileNotFoundError as e:
-            raise RuntimeError(f"Local registry cache not found. No file named {CACHE}.") from None
+        except FileNotFoundError:
+            raise RuntimeError(f"Local registry cache not found. No file named {CACHE_FNAME}.") from None
 
         try:
-            with open(CACHE_META, "r") as f:
+            with open(CACHE_META_FNAME, "r") as f:
                 data = json.load(f)
                 date = dateutil.parser.isoparse(data["last_cache_date"])
-        except (FileNotFoundError, ValueError) as e:
+        except (FileNotFoundError, ValueError):
             date = "Unknown"
 
         print(f"Registry loaded from cache dating: {date}")
 
     def _save_registry_as_cache(self):
-        cache_backup = CACHE + ".backup"
-        cache_meta_backup = CACHE_META + ".backup"
+        cache_backup = CACHE_FNAME + ".backup"
+        cache_meta_backup = CACHE_META_FNAME + ".backup"
+
+        # Create cache parent directories if they don't exist
+        for cache_dir in {os.path.dirname(CACHE_FNAME), os.path.dirname(CACHE_META_FNAME)}:
+            os.makedirs(cache_dir, exist_ok=True)
 
         # Suppressing a FileNotFoundError error for the first use case where a cached registry file
         # does not exist
         with contextlib.suppress(FileNotFoundError):
-            shutil.copy(CACHE, cache_backup)
-            shutil.copy(CACHE_META, cache_meta_backup)
+            shutil.copy(CACHE_FNAME, cache_backup)
+            shutil.copy(CACHE_META_FNAME, cache_meta_backup)
 
         try:
             metadata = {"last_cache_date": datetime.datetime.now(tz=datetime.timezone.utc).isoformat()}
 
             # Write the metadata
-            with open(CACHE_META, "w") as f:
+            with open(CACHE_META_FNAME, "w") as f:
                 json.dump(metadata, f)
 
             # write the registry
-            with open(CACHE, "w") as f:
+            with open(CACHE_FNAME, "w") as f:
                 json.dump(self._registry, f)
 
         except OSError as e:
-            # If either the cahce file or the metadata file could not be written, then restore from backup files
-            shutil.copy(cache_backup, CACHE)
-            shutil.copy(cache_meta_backup, CACHE_META)
+            # If either the cache file or the metadata file could not be written, then restore from backup files
```

Comment:
Is it easier to do this instead of reading the file content in memory and then just writing it back to the file if there is an error?

Reply:
I think it will be the same complexity. There isn't any benefit one way or another.